FirebaseAdmin/FirebaseAdmin/Auth/FirebaseUserManager.cs

```diff
@@ -68,27 +73,68 @@ public static FirebaseUserManager Create(FirebaseApp app)
         /// <param name="cancellationToken">A cancellation token to monitor the asynchronous
         /// operation.</param>
         /// <returns>A record of user with the queried id if one exists.</returns>
-        public async Task<UserRecord> GetUserById(
+        internal async Task<UserRecord> GetUserById(
             string uid, CancellationToken cancellationToken = default(CancellationToken))
         {
             if (string.IsNullOrEmpty(uid))
             {
                 throw new ArgumentException("User ID cannot be null or empty.");
             }
 
-            var payload = new Dictionary<string, object>()
+            var query = new UserQuery()
             {
-                { "localId", uid },
+                Field = "localId",
+                Value = uid,
+                Label = "uid",
             };
+            return await this.GetUserAsync(query, cancellationToken);
+        }
 
-            var response = await this.PostAndDeserializeAsync<GetAccountInfoResponse>(
-                "accounts:lookup", payload, cancellationToken).ConfigureAwait(false);
-            if (response == null || response.Users == null || response.Users.Count == 0)
+        /// <summary>
+        /// Gets the user data corresponding to the given email address.
+        /// </summary>
+        /// <param name="email">An email address.</param>
+        /// <param name="cancellationToken">A cancellation token to monitor the asynchronous
+        /// operation.</param>
+        /// <returns>A record of user with the queried email if one exists.</returns>
+        internal async Task<UserRecord> GetUserByEmail(
```

Comment:
This should be named GetUserByEmailAsync.

Reply:
Done